model/storage/grant/ClientCredentialsType.php

```diff
@@ -20,11 +20,26 @@
 
 namespace oat\taoOauth\model\storage\grant;
 
+use oat\taoOauth\model\provider\Provider;
+
 /**
  * Class ClientCredentialsType
  * @package oat\taoOauth\model\storage\grant
  */
 class ClientCredentialsType extends OauthCredentials
 {
     const NAME = 'client_credentials';
+
+    /**
+     * @return array
+     */
+    public function getProperties()
+    {
+        return array_merge(
+            parent::getProperties(),
+            [
+                Provider::SCOPE => !empty($this->properties[Provider::SCOPE]) ? $this->properties[Provider::SCOPE]: ''
```

Comment:
7.2 is here :)
$this->properties[Provider::SCOPE] ?? ''

Reply:
btw. do we need to show `SCOPE` field if it is empty to be backward compatible?

Reply:
If you leave it empty or if you ommit the `SCOPE` parameter then is up to the server to respond with a default value for it (if it takes into account). 
And by the specification, the response should include also the `scope` which access was granted because it can be different from the requested. 

i don't think there is problem of showing if empty.